pandas/core/arrays/base.py

```diff
@@ -93,7 +93,9 @@
 
 if TYPE_CHECKING:
 
-    class ExtensionArraySupportsAnyAll("ExtensionArray"):
+    class ExtensionArraySupportsAnyAll(
+        "ExtensionArray"
+    ):  # pylint: disable=inherit-non-class
```

Comment:
does it work to do
```
class ExtensionArraySupportsAnyAll(ExtensionArray)
```
?

Reply:
I'm getting flake8 errors for `pandas/core/arrays/base.py:96:40: F821 undefined name 'ExtensionArray'`.

Reply:
This might be a dumb question but is it because `ExtensionArray` is defined below `ExtensionArraySupportsAnyAll`?

Reply:
Thanks for checking, you're right

Perhaps we can move the definition of `ExtensionArraySupportsAnyAll` to right after the definition of `ExtensionArraySupportsAnyAll`, i.e.

So
```
class ExtensionArray:
    ...

class ExtensionArraySupportsAnyAll(ExtensionArray):
    def any(self, *, skipna: bool = True) -> bool:
        pass

    def all(self, *, skipna: bool = True) -> bool:
        pass
```
?

I'll check again tomorrow if that's fine

Reply:
Of course. I'll move it and run the pre-commit and see how it goes.

Reply:
Have moved it below `ExtensionArray`. Passed all pre-commits. Just waiting on CI

Reply:
cool, seems fine, thanks